src/test/AludelV3.t.sol

```diff
@@ -296,4 +270,359 @@ contract AludelV3Test is Test {
         assertEq(data.totalStake, STAKE_AMOUNT * 2);
         assertEq(data.lastUpdate, block.timestamp);
     }
+
+    function test_single_unstake(uint8 schedules, uint40 scheduleDuration, uint256 rewardAmount, uint256 stakingAmount)
+        public
+    {
+        vm.assume(schedules > 0);
+        vm.assume(scheduleDuration > 0);
+
+        // I'm capping the amounts to avoid shares overflow
+        stakingAmount = bound(stakingAmount, 1, 1_000_000_000_000 ether);
+        rewardAmount = bound(rewardAmount, 1, 1_000_000_000_000 ether);
+
+        Crucible crucible = Utils.createCrucible(user, crucibleFactory);
+        Utils.fundMockToken(address(crucible), stakingToken, stakingAmount * schedules);
+
+        AludelV3.AludelData memory data = aludel.getAludelData();
+
+        Utils.stake(user, crucible, aludel, stakingToken, stakingAmount);
+
+        for (uint256 i = 0; i < schedules; i++) {
+            Utils.fundAludel(aludel, admin, rewardToken, rewardAmount, scheduleDuration);
+            vm.warp(block.timestamp + scheduleDuration);
+        }
+
+        uint256[] memory indices = new uint256[](1);
+        uint256[] memory amounts = new uint256[](1);
+        indices[0] = 0;
+        amounts[0] = stakingAmount;
+
+        Utils.unstake(user, crucible, aludel, stakingToken, indices, amounts);
+    }
+
+    function test_fund_without_approval(uint64 reward, uint32 duration) public {
+        vm.assume(reward > 0);
```

Comment:
we have a test for a duration of zero but not for a reward of zero

Reply:
That's because we allow to fund 0 tokens.
I created an issue (#57) to track it.